### ISSUE TITLE
Parallel execution for tasks

### ISSUE THREAD

**Issue by erdincka** [hpe-container-platform-community/ezdemo#14]
Some tasks can run in parallel, ie, "add workers" and "configure DF node", "create k8scluster" and "create DF". 
Currently running Samba in docker container within ad_server is started as async task, and is never checked for completion/errors.  Would be nice to have a method to submit tasks in the background, and check/wait just before the job that depends it (create tenant should check create cluster, install mapr should check AD integration etc). 
Ansible has async method, and I guess it is the best option to implement this (need to ensure we submit and check these jobs on the same hosts etc).
